.bitmap

```diff
@@ -525,6 +525,36 @@
         "mainFile": "index.ts",
         "rootDir": "scopes/ui-foundation/harmony-ui-app/harmony-ui-app"
     },
+    "hooks/lanes-context": {
+        "scope": "",
+        "version": "",
+        "mainFile": "index.ts",
+        "rootDir": "scopes/lanes/hooks/lanes-context"
+    },
+    "hooks/lanes-provider": {
+        "scope": "",
+        "version": "",
+        "mainFile": "index.ts",
+        "rootDir": "scopes/lanes/hooks/lanes-provider"
+    },
```

Comment:
wouldn't separate the provider and the context. would do a single component names lanes-provider - would put them under the same component. It is also important for usability but also might cause technical issues for different versions of the context.

Reply:
The reason I split them is because,

- useLanes abstracts whether you are calling the context or the GQL API, that means useLanes has a dependency on `LanesContext` since it will call the context first, if it doesn’t get the data, it will call the GQL API

- `LanesProvider` uses useLanes to hydrate the `LanesContext` . It accepts a `viewedLaneId` (which is a client side only state) and hydrates the `LanesModel` from GQL response.

- Since `useLanes` has a dependency on the `LanesContext` , combining `LanesContext` and `LanesProvider` would result in a cyclic dependency

Reply:
combined, `useLanes`, `LanesContext`, and `LanesProvider` into `useLanes`